### DISCUSSION TITLE
Viewing CHM files

### DISCUSSION THREAD

**Discussion by crustyturd** [sumatrapdfreader/sumatrapdf#2570] [General]
I have a 4k screen. With current pre-releases viewing CHM files has become much more difficult. If 'UseFixedPageUI' = 'false' for 'ChmUI' setting, the font is too small to see. Having to zoom in for every file is very time consuming and annoying. Also, 'UseFixedPageUI' = 'false' is the only way to wrap text when zooming.
If 'UseFixedPageUI' = 'true', the font size is larger (not as large as I would like), but zooming does not wrap text.
I would like to see a default zoom level percentage for CHM files, if possible, or, some other solution to alleviate this problem.

**Comment by kjk** [comment]
Do you mean that it's different than in latest released version (3.3.3) ?

Could you post a screenshot of the difference between 3.3.3 (you can use portable version) and pre-release?

I don't think I've made any changes to how CHM files are displayed.

**Reply by kjk** [reply]
I've made a change to adjust the zoom for CHM by DPI. Please try pre-release 14587 or later and let me know if it fixes it for you.

**Reply by GitHubRulesOK** [reply]
You may find you cant please all of the people 🙂 all of the 🕐 
For me there were two issues one was the undersized aspect of both modes of which one is resolved
So independent of settings dpi the replacement rendering of chm reader is now better at any dpi,
 the second is dependent on custom dpi (was font) which for image scaling I keep at 96 dpi
 hence the smaller top right render the alternative for me is to set dpi to 192 (96 X 2) for scale fidelity
 but then chm reader is too big (however I use rarely thus suppose I could (just for CHM) temporarily make an edit but its fiddly.
![image](https://user-images.githubusercontent.com/12754913/168424471-4e6b0ad2-5a6b-4282-b019-fbf724192f69.png)

**Reply by crustyturd** [reply]
The latest pre-release does not start at all. I get the hour glass for a couple of seconds, then nothing.
Nothing shows up in task manager.

**Reply by crustyturd** [reply]
Forgot to mention latest prerel is 14590. Cannot find 14587.

**Reply by crustyturd** [reply]
Here is from EvenLog Viewer after trying to start SumatraPDF.exe:
Faulting application name: SumatraPDF-prerel-64.exe, version: 3.4.0.0, time stamp: 0x627f711d
Faulting module name: SumatraPDF-prerel-64.exe, version: 3.4.0.0, time stamp: 0x627f711d
Exception code: 0xc0000005
Fault offset: 0x000000000003be52
Faulting process id: 0x2a4

**Reply by kjk** [reply]
sorry about that, hopefully will be fixed 14591 (should be available ~1hr)

**Reply by crustyturd** [reply]
I know you can't please all the people all the time, but thanks for trying.

**Reply by kjk** [reply]
ok, 14591 is available and seems I fixed the crash

**Reply by crustyturd** [reply]
Yes, now it loads. However, It is only marginally larger. Also, in this build it seems that the document color is not inverted when pressing 'i' key. It is still inverted for the sidebar and for PDF documents.
Anyway, thanks for trying. This is a great program and I will continue to use it.

**Reply by crustyturd** [reply]
Sorry, it appears invert colors does not work only when CHM 'UseFixedPageUI=false'. I guess it's supposed to work that way.

**Reply by kjk** [reply]
For what it's worth, with this change now I get the exact same sizing in SumatraPDF as in Windows'  HTML Help

**Reply by crustyturd** [reply]
Yes, here too. I was looking for some improvement over that. For now, I will keep using an earlier pre-release that still uses EBook FontName and FontSize. I don't get the word wrap if I zoom in, but I can at least set a font size such that I rarely need to zoom. Until, that is, you add a new 'must have' feature. Then I will have to reconsider.
Thanks again for spending so much time on this.

**Comment by GitHubRulesOK** [comment]
@kjk 
I don't see any change between 3.1.2 and current 3.4.14583 however SumatraPDF in both modes is noticeably smaller than MS
here using standard HD (1920x1080) at 125% (recommended)
![image](https://user-images.githubusercontent.com/12754913/168385353-c2bc2ac4-615d-426b-a7c2-6b39fe8ce4b6.png)

**Reply by crustyturd** [reply]
I'm sorry, I should have mentioned my 4k screen is set to 3840x2160 UHD with system scaling at 200%. I also use high-contrast mode.
Actually, they look the same with 3.3.3 as with pre-release. However, in 3.3.3 I was able to set EBook font and size which
applied to CHM files when UseFixedPageUI=true. I'd rather not use compatiblity mode scaling because it looks much better in native mode, but it appears much smaller of course. If I cannot set default zoom for CHM files, I will try compatibility mode scaling to system (200%), which is OK but not as good as native mode.
Here is screenshot at 3840x2160, UseFixedPageUI=false:
![sshot-002](https://user-images.githubusercontent.com/83983916/168403417-9f89e4e3-6670-460e-b3ab-8eaa6cd83910.png)
